src/lib/ifaces/iface.rs

```diff
@@ -227,6 +233,11 @@ pub(crate) fn parse_nl_msg_to_iface(nl_msg: &LinkMessage) -> Option<Iface> {
                         IfaceType::Bridge => {
                             iface_state.bridge = get_bridge_info(&d)
                         }
+                        IfaceType::Tun => {
+                            if let Ok(info) = get_tun_info(&d) {
```

Comment:
You are ignoring the failure found by `get_bridge_info()`. Please at least use `eprintln!()` to show the errors.

Reply:
How is it now?